src/file.cr

```diff
@@ -377,6 +378,15 @@ class File < IO::FileDescriptor
     stat.size
   end
 
+  # Truncates the file to the specified length. Requires a write file descriptor
+  def truncate(length)
```

Comment:
Yes, sorry I didn't follow up, I've had a couple of very busy days (and tomorrow will be as well). I was playing with specifying a size and flushing the buffer and I'm still a bit confused by the behavior, I've seen some inconsistent results like the one I mentioned (even flushing). I wanted to play a bit more until I understand it and can be confident that the method works as expected, I'll do so in the next couple of days.

Thanks for the feedback to both!